packages/vector_graphics_compiler/lib/src/svg/parsers.dart

```diff
@@ -140,3 +141,19 @@ double parsePercentage(String val, {double multiplier = 1.0}) {
 
 /// Whether a string should be treated as a percentage (i.e. if it ends with a `'%'`).
 bool isPercentage(String val) => val.endsWith('%');
+
+/// Parses value from the form '25%', 0.25 or 25.0 as a double.
+/// Note: Percentage or decimals will be multipled by the total
+/// view box size, where as doubles will be returned as is.
+double? parsePatternUnitToDouble(String rawValue, {ViewportNode? viewBox}) {
```

Comment:
Could you add some unit tests that invoke this method directly with different values?

Reply:
done.